wgsl/index.bs

```diff
@@ -422,6 +422,147 @@ The result type depends on the number of letters provided. Assuming a `vec4<f32>
   </xmp>
 </div>
 
+# Textures # {#textures}
+
+## Texture Types ## {#texture-types}
+
+### Sampled Texture ### {#texture-sampled}
+
+<pre class='def'>
+`texture_sampled_1d<type>`
+  %1 = OpTypeImage %type 1D 0 0 0 1 Unknown
+
+`texture_sampled_1d_array<type>`
+  %1 = OpTypeImage %type 1D 0 1 0 1 Unknown
+
+`texture_sampled_2d<type>`
+  %1 = OpTypeImage %type 2D 0 0 0 1 Unknown
+
+`texture_sampled_2d_array<type>`
+  %1 = OpTypeImage %type 2D 0 1 0 1 Unknown
+
+`texture_sampled_3d<type>`
+  %1 = OpTypeImage %type 3D 0 0 0 1 Unknown
+
+`texture_sampled_2d_ms<type>`
+  %1 = OpTypeImage %type 2D 0 0 1 1 Unknown
+
+`texture_sampled_2d_ms_array<type>`
+  %1 = OpTypeImage %type 2D 0 1 1 1 Unknown
+
+`texture_sampled_cube<type>`
+  %1 = OpTypeImage %type Cube 0 0 0 1 Unknown
+
+`texture_sampled_cube_array<type>`
+  %1 = OpTypeImage %type Cube 0 1 0 1 Unknown
+</pre>
+* type must be `f32`, `i32` or `u32`
+* The parameterized type for the images is the type after conversion from sampling.
+    E.g. you can have an image with texels with 8bit unorm components, but when you sample
+    them you get a 32-bit float result (or vec-of-f32).
+
+### Read-only Storage Texture ### {#texture-ro}
+<pre class='def'>
+`texture_ro_1d<type, image_storage_type>`
+  %1 = OpTypeImage %type 1D 0 0 0 2 image_storage_type
+
+`texture_ro_1d_array<type, image_storage_type>`
+  %1 = OpTypeImage %type 1D 0 1 0 2 image_storage_type
+
+`texture_ro_2d<type, image_storage_type>`
+  %1 = OpTypeImage %type 2D 0 0 0 2 image_storage_type
+
+`texture_ro_2d_array<type, image_storage_type>`
+  %1 = OpTypeImage %type 2D 0 1 0 2 image_storage_type
+
+`texture_ro_3d<type, image_storage_type>`
+  %1 = OpTypeImage %type 3D 0 0 0 2 image_storage_type
+</pre>
+* type must be `f32`, `i32` or `u32`
+* The parameterized type for the images is the type after conversion from reading.
+    E.g. you can have an image with texels with 8bit unorm components, but when you read
+    them you get a 32-bit float result (or vec-of-f32).
+
+### Write-only Storage Texture ### {#texture-wo}
+<pre class='def'>
+`texture_wo_1d<image_storage_type>`
+  %1 = OpTypeImage %void 1D 0 0 0 2 image_storage_type
+
+`texture_wo_1d_array<image_storage_type>`
+  %1 = OpTypeImage %void 1D 0 1 0 2 image_storage_type
+
+`texture_wo_2d<image_storage_type>`
+  %1 = OpTypeImage %void 2D 0 0 0 2 image_storage_type
+
+`texture_wo_2d_array<image_storage_type>`
+  %1 = OpTypeImage %void 2D 0 1 0 2 image_storage_type
+
+`texture_wo_3d<image_storage_type>`
+  %1 = OpTypeImage %void 3D 0 0 0 2 image_storage_type
+</pre>
+
+### Depth Texture ### {#texture-depth}
```

Comment:
where did the cube and cube_array go?

Reply:
So, depth should have cube and cube_array? Though from previous comment it was only sampled. So, Sampled and Depth should have cube and cube array, read/write should not, yea?

Reply:
correct

Reply:
Done